samples/data-sealing/CMakeLists.txt

```diff
@@ -0,0 +1,77 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+cmake_minimum_required(VERSION 3.10)
+
+project("Data Sealing Sample" LANGUAGES C CXX)
+
+find_package(Threads REQUIRED)
+
+find_package(OpenEnclave CONFIG REQUIRED)
+add_executable(openenclave::oeedger8r IMPORTED)
+set_target_properties(openenclave::oeedger8r PROPERTIES IMPORTED_LOCATION ${OE_BINDIR}/oeedger8r)
+
+add_custom_command(OUTPUT ${CMAKE_SOURCE_DIR}/common/datasealing_t.c
+  COMMAND openenclave::oeedger8r --trusted ${CMAKE_SOURCE_DIR}/datasealing.edl
+  WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}/common)
+
+# TODO: Move to subdirectories after refactoring
+
+# Enclave 1
+add_executable(enclave_a_v1
+  enc1/ecalls.cpp
+  common/dispatcher.cpp
+  common/keys.cpp
+  common/datasealing_t.c)
+
+target_link_libraries(enclave_a_v1 openenclave::oeenclave openenclave::oelibcxx)
+
+add_custom_command(OUTPUT ${CMAKE_SOURCE_DIR}/enc1/private.pem ${CMAKE_SOURCE_DIR}/enc1/public.pem
```

Comment:
Similarly for the generation of the private keys, it's probably better to have `private_a.pem` and `private_b.pem` generated as part of `common`, and then have the enclave builds depend on those.

I would not remove the Makefiles build, but they should be updated to be consistent with the CMake one.

Reply:
Again, I agree, but this was the implementation _without_ changing the existing build or project structure.